__tests__/classes/CanvasRenderingContext2D.__getClippingPath.js

```diff
@@ -51,4 +51,17 @@ describe('__getClippingRegion', () => {
     ctx.restore();
     expect(region).toStrictEqual(ctx.__getClippingRegion());
   });
+
+  it('should delete current clipping region when restored', () => {
+    ctx.rect(1, 2, 3, 4);
+    ctx.clip();
+    ctx.save();
+    ctx.rect(1, 2, 3, 4);
+    ctx.arc(1, 2, 3, 4, 5);
+    ctx.clip();
+    const region = ctx.__getClippingRegion();
```

Comment:
First, I want to admit that this test function does indeed assert that the `restore()` function works in relation to clip calls. Thank you.

With that being said, I believe simplifying the tests here will go a long way in describing how the functions are supposed to work.

If possible, I would like to remove a few of the function calls in this test, so I will outline the reasons why I suggest them.

1. On line 62, this code obtains the current clipping region, which normally would be able to be used, except that the `afterEach()`  callback already performs clipping region checks. Thus, we don't have to obtain the clipping region at all.
2. On line 64, this code calls `ctx.save()` again, which becomes a confounding factor when proving that the `ctx.restore()` function works. We should probably remove it to make things clearer.
3. Lastly, on line 65 creating another expectation here is not necessary, because we know the clipping region should only contain a single `Rect` entry at this point.

All in all, great work. Please remove the lines in question, and I will accept the contribution. Thank you so much for your effort.

Reply:
Thanks for working with me on this. I haven't been working with javascript long.

I tried removing lines 62 and 64-65, but then the test passes even when `this._clipStack.pop()` is not inside the `restore()` implementation, which doesn't seem right to me. I think the way to ensure that the clipping region has been popped off the stack is by calling `ctx.save() -> ctx.restore() -> ctx.save()` and confirming that the clipped region immediately after the two `ctx.save()` calls do not have the same identity.

The `afterEach()` snapshot check already occurs in a [different test](https://github.com/hustcc/jest-canvas-mock/blob/master/__tests__/classes/CanvasRenderingContext2D.__getClippingPath.js#L51), which is what I thought you meant when you said to consolidate earlier.

Is there a better way to check that the pop happens?

Reply:
Okay. At the very least, we should make sure not use expect calls here. We can just inspect the state after the test runs, and that's enough.

As for your example, I think I'm being pedantic.

Just remove the expect calls from these tests and we should be good.

Reply:
There's an additional test that doesn't require expect calls too. I can understand why they were added in the first place. They aren't necessary.

Reply:
That makes sense. Fixed.